internal/service/alarms/internal/alarms_server.go

```diff
@@ -190,9 +195,110 @@ func (a *AlarmsServer) GetAlarm(ctx context.Context, request api.GetAlarmRequest
 	return api.GetAlarm200JSONResponse(models.ConvertAlarmEventRecordModelToApi(*record)), nil
 }
 
-func (a *AlarmsServer) AckAlarm(ctx context.Context, request api.AckAlarmRequestObject) (api.AckAlarmResponseObject, error) {
-	// TODO implement me
-	return nil, fmt.Errorf("not implemented")
+// PatchAlarm handles an API request to patch an Alarm Event Record
+func (a *AlarmsServer) PatchAlarm(ctx context.Context, request api.PatchAlarmRequestObject) (api.PatchAlarmResponseObject, error) {
```

Comment:
As discussed previously and noted in the design doc...we are only supporting Ack. i,e user may only supply AlarmEventRecordModifications with a alarmAcknowledged bool. Let's revert the spec and follow the design as agreed upon.

Reply:
I don´t mind reverting this part. However, I thought this is a good opportunity to put this into discussion again. So far, we've been making an effort to follow what the specification defines. Not only what is defined in O-RAN.WG6.O2IMS-INTERFACE-R004-v07.00 but also properties of fields defined in O-RAN.WG6.O2IMS-INTERFACE-R004-v07.00 (format, nullable, etc).

The current version (supporting only ACK) does not follow at all what is defined in table 3.3.6.2.4-1 (O-RAN.WG6.O2IMS-INTERFACE-R004-v07.00). Not only the lack of the `perceivedSeverity` field but also that these are nullable fields. @alegacy has been running conformance tests. It seems there are none for this, but eventually they will. I don't see why we cannot follow the specification.

Another thing, the initial design is not set in stone (not even merged). Luckily, we are still in the initial phases. We could change stuff as we progress with the implementation.

As said, I don't mind changing it back, but I think it is wrong. Let's see what @alegacy and/or @browsell thinks.

Reply:
The reason why we landed on only ack (for now at least) is that we cant modify the caas alerts data. The PerceivedSeverity is essentially PromRule label `severity` which is hardcoded.

Reply:
@pixelsoccupied I must have missed the conversation about why we wouldn't support the clear use case.  AFAIK, our goal is to try to support as much of the spec as technically possible so unless there is a strong reason why we shouldn't I think we should proceed with both the ack and clear.   Of course, we'll have to be mindful of the manually cleared state when syncing alarms in from downstream sources so that we don't un-clear it again when we sync them in.

Reply:
@browsell @pixelsoccupied  You are correct that we only have read-only access to the source data, but given that we are persisting an intermediate copy there are some things that we can do to support the API requirements. ... we just have to be careful not to overwrite any user modifications if we re-sync the data.  Of course, if there is a technical reason why we can't do that then this is a problem, but is that the case?

Reply:
@alegacy @pixelsoccupied @browsell 

This is from the use case for reference to support my previous message about Manual and Automatic type of alarms

```
This use case is used for a management entity to acknowledge or request for alarm(s) to be cleared. It is also used for
the IMS to clear alarm(s) autonomously. It is expected that some resources could have some alarms that can be
manually cleared while other alarms would be automatically cleared. 
```

Reply:
Discussed internally. We will follow proposed approach for clearing alarms. Distinction between Manual and Automatic ones.